template-sample-app/template/src/config/store.js

```diff
@@ -76,6 +76,15 @@ export default function configureAppStore() {
   const reducerManager = createReducerManager(staticReducers);
   store = configureStore({
     reducer: reducerManager.reduce,
+    middleware: getDefaultMiddleware({
```

Comment:
I see that we are still trying to force mutations to the store... Didn't we consider any other option to avoid this? (cc/ @padawannn) 

As a side note, we should check if we can customize a bit the middleware configuration depending on the environment. Reading https://github.com/leoasis/redux-immutable-state-invariant, from which the immutability middleware is ported, it says it should be removed from production bundle, to improve perf.